pkg/cloudprovider/cloudprovider.go

```diff
@@ -87,6 +87,9 @@ func (c *CloudProvider) Create(ctx context.Context, machine *v1alpha5.Machine) (
 	if err != nil {
 		return nil, fmt.Errorf("resolving node template, %w", err)
 	}
+	machine.Annotations = lo.Assign(machine.Annotations, map[string]string{
```

Comment:
Should this be resolved at the end with all the other `toMachine` details?

Reply:
No, we should not because we want to only resolve this on creation. `toMachine` is utilized on GET and LIST calls

Reply:
I don't think this value gets propagated back to the returned Machine if you do it this way. You should validate this with E2E testing but I'm surprised if it is.